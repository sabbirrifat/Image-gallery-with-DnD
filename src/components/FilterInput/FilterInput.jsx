import React from "react";
import "./filter-input.styles.css";

const FilterInput = ({ name, icon, value, handleChange }) => {
  return (
    <div className="filter">
      <div className="slider-left">
        <img src={icon} alt="" />
      </div>

      <input
        name={name}
        type="range"
        min="0"
        max="100"
        data-testid={name}
        value={value}
        onChange={handleChange}
      />

      <div className="slider-right">{`${value} ${
        name === "saturation" ? "dmp" : "%"
      }`}</div>
    </div>
  );
};

export default FilterInput;
