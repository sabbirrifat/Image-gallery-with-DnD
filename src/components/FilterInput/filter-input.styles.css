.filter {
  display: flex;
  align-items: center;
  width: 100%;
  margin: 14px auto;
}

.filter input {
  -webkit-appearance: none;
  height: 6px;
  flex: 1;
  background: #a1a6aa;
  border-radius: 5px;
  outline: none;
  border: none;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
}

.filter input::-webkit-slider-thumb {
  -webkit-appearance: none;
  height: 20px;
  width: 20px;
  background: white;
  border: 2px solid #a1a6aa;
  border-radius: 50%;
  cursor: pointer;
  -webkit-border-radius: 50%;
  -moz-border-radius: 50%;
  -ms-border-radius: 50%;
  -o-border-radius: 50%;
}

.filter .slider-right {
  margin-left: 14px;
  font-size: 14px;
  color: #a1a6aa;
}

.filter .slider-left {
  margin-right: 14px;
}
