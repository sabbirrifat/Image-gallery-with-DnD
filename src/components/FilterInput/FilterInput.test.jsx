import FilterInput from "./FilterInput";
import React from "react";
import { render, cleanup } from "@testing-library/react";

const handleChange = (event) => {};
const [name, value, icon] = ["blur", "20", "https://test.com/123.jpg"];

const mockFilterInput =
  '<input name="blur" type="range" min="0" max="100" data-testid="blur" value="20">';

describe("Filter Input Test", () => {
  afterEach(cleanup);

  it("create a snapshop", () => {
    expect(
      render(
        <FilterInput
          name={name}
          value={value}
          icon={icon}
          handleChange={handleChange}
        />
      )
    ).toMatchSnapshot();
  });

  it("should match the mock output", () => {
    const { getByTestId } = render(
      <FilterInput
        name={name}
        value={value}
        icon={icon}
        handleChange={handleChange}
      />
    );
    expect(getByTestId("blur").outerHTML).toEqual(mockFilterInput);
  });

  it("should match the image src", () => {
    const { getByRole } = render(
      <FilterInput
        name={name}
        value={value}
        icon={icon}
        handleChange={handleChange}
      />
    );
    expect(getByRole("img").src).toEqual("https://test.com/123.jpg");
  });
});
