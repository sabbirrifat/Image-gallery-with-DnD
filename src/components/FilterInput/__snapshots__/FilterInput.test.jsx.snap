// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Input Test create a snapshop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter"
      >
        <div
          class="slider-left"
        >
          <img
            alt=""
            src="https://test.com/123.jpg"
          />
        </div>
        <input
          data-testid="blur"
          max="100"
          min="0"
          name="blur"
          type="range"
          value="20"
        />
        <div
          class="slider-right"
        >
          20 %
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter"
    >
      <div
        class="slider-left"
      >
        <img
          alt=""
          src="https://test.com/123.jpg"
        />
      </div>
      <input
        data-testid="blur"
        max="100"
        min="0"
        name="blur"
        type="range"
        value="20"
      />
      <div
        class="slider-right"
      >
        20 %
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
